Implement server endpoints

edan-bainglass opened the pull request:
Supersedes #76

This PR migrates the /server/* endpoints from the old restapi.

Closes #117

edan-bainglass committed:
Implement server router

eimrek reviewed (changes requested):
Looks good, some small changes. Also `/graphql` currently doesn't appear under the API_CONFIG['PREFIX'] nor listed in the `/docs`.

eimrek commented on aiida_restapi/routers/server.py:
```diff
@@ -0,0 +1,204 @@
+    API_minor_version: str = pdt.Field(description='Minor version of the API')
+    API_revision_version: str = pdt.Field(description='Revision version of the API')
+    API_prefix: str = pdt.Field(description='Prefix for all API endpoints')
+    AiiDA_version: str = pdt.Field(description='Version of the AiiDA installation')
```
I think we should go for `snake_case` for the field names here. E.g. `aiida_version`, not sure what else to say. No need to port the old rest api exactly.

eimrek commented on aiida_restapi/config.py:
```diff
@@ -18,3 +20,8 @@
 }
+
+API_CONFIG = {
+    'PREFIX': '/api/v0',
```
Fine to change old behavior (we're anyways breaking it).

I think the `/v0` prefix could be kept, AFAIK, one could potentially implement/expose multiple versions of the api in the same FastAPI process.

edan-bainglass committed:
Use snake case for server info

edan-bainglass committed:
Drop `/api` from prefix

edan-bainglass committed:
Serve `/graphql` at configured prefix

edan-bainglass commented:
> Looks good, some small changes. Also `/graphql` currently doesn't appear under the API_CONFIG['PREFIX'] nor listed in the `/docs`.

Thanks @eimrek. I've addressed your comments. `/graphql` is now also served at the app prefix. However, the current implementation is incompatible with FastAPI's auto-openapi construction, hence no immediate support for docs and redoc. Looking online, it appears that the implementation is also possibly using an old style. The GraphQL module will be properly addressed in a later PR.

edan-bainglass commented:
Hi @eimrek. This is now updated. Can you please re-review? 🙏

eimrek reviewed (approved):
👍

edan-bainglass committed:
Add API prefix to `/node/types` endpoint
